content/blog/actions-2-0-0/index.Rmd

```diff
@@ -0,0 +1,155 @@
+
+photo:
+  url: https://www.pexels.com/photo/industry-technology-power-clock-4069389
+  author: Ronaldo Galeano
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [programming] 
+tags: ["GitHub Actions", "r-lib", "actions", "CI", "contionuous integration"]
+---
+
+```{=html}
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+```
+We're tickled pink to announce the new, `v2` versions of our collection of R related GitHub Actions at <https://github.com/r-lib/actions>.
+
+If you are already using these actions, you might want to take look at the [full list of changes](https://github.com/r-lib/actions/releases/tag/v2) first.
+
+## About `rlib/actions`
+
+[GitHub Actions](https://github.com/features/actions) is a continuous integration service that allows you to automatically run code whenever you push to GitHub.
+If you're developing a package this allows you to automate tasks like running `R CMD check` on multiple platforms or rebuilding your [pkgdown](https://pkgdown.r-lib.org/) website.
+
+The [`r-lib/actions`](https://github.com/r-lib/actions#readme) repo has a number of reusable actions that perform common R-related tasks: installing R and Rtools, pandoc, installing dependencies of R packages, running `R CMD check`, etc.:
+
+-   `setup-r` installs R and on Windows Rtools,
+
+-   `setup-pandoc` installs pandoc,
+
+-   `setup-r-dependencies` installs R package dependencies,
+
+-   `check-r-package` runs `R CMD check` on an R package.
+
+See the [README](https://github.com/r-lib/actions#readme) for the complete list of actions.
+
+## Setting up `r-lib/actions`
+
+The `r-lib/actions` repo has [example workflows](https://github.com/r-lib/actions/tree/v2-branch/examples#example-workflows), it is best to start with these.
+
+You can copy the ones you'd like to use to the `.github/workflows` directory of your R package or project.
+For an R package you would typically want the `test-coverage` workflow and one of the `check-` workflows, depending on how thoroughly you want to check your package across operating systems and R versions.
+If your package has a pkgdown site then you probably also want the `pkgdown` workflow.
+
+The usethis package has several helper functions to set up GitHub Actions for you: `?usethis::use_github_action`.
+Note, that to set up the new `v2` versions of the actions you currently need to use the [development version](https://github.com/r-lib/usethis#installation) of the usethis package:
```

Comment:
I think it would be good to get a usethis patch release out before this blog post. Do you think that would be possible @jennybc?

Reply:
No, if we want to merge / post this ASAP.
But yes if it's less urgent.

I think I should reach a reasonable stopping point in readxl by midweek and could work on a usethis patch release, which could maybe be submitted one week from today?

Reply:
The post is not particularly urgent, I think.

Reply:
OK let's let this age for several days. I would like to get a patch release of usethis out anyway and this is a good forcing function.